metadata-ingestion/src/datahub/emitter/rest_emitter.py

```diff
@@ -329,6 +330,8 @@ def __init__(
         self._openapi_ingestion = (
             openapi_ingestion  # Re-evaluated after test connection
         )
+        self._config_ttl_seconds = config_ttl_seconds
+        self._config_fetch_time: Optional[float] = None
```

Comment:
Have you considered using the TTL cache decorator in https://pypi.org/project/cachetools/ library ?
It might help simplify some of this boilerplate code.